webapp/components/Modal/ReportModal.vue

```diff
@@ -79,18 +76,24 @@ export default {
           `,
           variables: { id: this.id },
         })
-        this.success = true
         this.$toast.success(this.$t('report.success'))
-        setTimeout(() => {
-          this.isOpen = false
-          setTimeout(() => {
-            this.success = false
-            this.$emit('close')
-          }, 500)
-        }, 1500)
+        this.$emit('close')
+        this.isOpen = false
+        this.success = false
       } catch (err) {
+        this.$emit('close')
         this.success = false
-        this.$toast.error(err.message)
+        switch (err.message) {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-21T12:56:31Z" title="Friday, June 21st 2019, 2:56:31 pm +02:00">Jun 21, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -79,18 +78,25 @@ export default {
           `,
           variables: { id: this.id },
         })
-        this.success = true
+        // this.success = true
         this.$toast.success(this.$t('report.success'))
-        setTimeout(() => {
-          this.isOpen = false
-          setTimeout(() => {
-            this.success = false
-            this.$emit('close')
-          }, 500)
-        }, 1500)
+        this.$emit('close')
+        this.isOpen = false
+        this.success = false
       } catch (err) {
+        this.$emit('close')
         this.success = false
-        this.$toast.error(err.message)
+        switch (err.message) {
```


In this case I would suggest a generic message like: `Reporting the resource was not successful`  and get rid of the `switch` statement. Feel free to keep the switch if you really think the different messages give better user experience.